### ISSUE TITLE
Packer.io inside of Vagrant instance fails..

### ISSUE THREAD

**Issue by diranged** [mitchellh/packer#1767]
I'm trying to run Packer to create an extremely simple docker container from within a Vagrant instance on my Macbook. Here's the setup ... start with Vagrant/Vbox and this Vagrantfile:

#####  Vagrantfile:
```
Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu"

  config.vm.provider :virtualbox do |vb|
    vb.name = 'packer-builder'
    vb.customize ["modifyvm", :id, "--memory", "2048" ]
    vb.customize ["modifyvm", :id, "--cpus",  "2" ]
    vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
  end

  if Dir.glob("#{File.dirname(__FILE__)}/.vagrant/machines/default/*/id").empty?
    pkg_cmd = "wget -q -O - https://get.docker.io/gpg | apt-key add -; "
    pkg_cmd << "echo deb http://get.docker.io/ubuntu docker main > /etc/apt/sources.list.d/docker.list;"
    pkg_cmd << "apt-get update -qq; apt-get install -q -y --force-yes lxc-docker bsdtar; "
    pkg_cmd << "usermod -a -G docker vagrant; "
    pkg_cmd << "cd /usr/bin; wget -qO- https://dl.bintray.com/mitchellh/packer/packer_0.7.5_linux_amd64.zip | bsdtar -pxvf-; "
    pkg_cmd << "sudo chmod +x /usr/bin/packer*; "
    config.vm.provision :shell, :inline => pkg_cmd
  end
end
```

Then setup Vagrant ...
##### vagrant up
```
Matts-MacBook-Pro:packer-test diranged$ vagrant up 
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'ubuntu'...
==> default: Matching MAC address for NAT networking...
==> default: Setting the name of the VM: packer-builder
...
...
==> default: x packer-provisioner-shell
```

Now, using this very simple json file, try to run packer...

##### simple.json
```json5
{ "builders": [
    {
      "type": "docker",
      "image": "ubuntu",
      "commit": false
    }
  ],

  "provisioners": [
    { "type": "shell", "inline": [ "sleep 30" ] }
  ]
}
```

##### PACKER_LOG=1 packer build simple.json
```
vagrant@vagrant-ubuntu-trusty-64:/vagrant$ !PA
PACKER_LOG=1 packer build simple.json 
2014/12/16 05:41:53 [INFO] Packer version: 0.7.5  9cd66feeacbd9cb318b72eb5ed59428c5b8c37ac+CHANGES
2014/12/16 05:41:53 Packer Target OS/Arch: linux amd64
2014/12/16 05:41:53 Built with Go Version: go1.3.3
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: amazon-chroot = /usr/bin/packer-builder-amazon-chroot
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: amazon-ebs = /usr/bin/packer-builder-amazon-ebs
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: amazon-instance = /usr/bin/packer-builder-amazon-instance
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: digitalocean = /usr/bin/packer-builder-digitalocean
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: docker = /usr/bin/packer-builder-docker
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: googlecompute = /usr/bin/packer-builder-googlecompute
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: null = /usr/bin/packer-builder-null
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: openstack = /usr/bin/packer-builder-openstack
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: parallels-iso = /usr/bin/packer-builder-parallels-iso
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: parallels-pvm = /usr/bin/packer-builder-parallels-pvm
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: qemu = /usr/bin/packer-builder-qemu
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: virtualbox-iso = /usr/bin/packer-builder-virtualbox-iso
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: virtualbox-ovf = /usr/bin/packer-builder-virtualbox-ovf
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: vmware-iso = /usr/bin/packer-builder-vmware-iso
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: vmware-vmx = /usr/bin/packer-builder-vmware-vmx
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: atlas = /usr/bin/packer-post-processor-atlas
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: compress = /usr/bin/packer-post-processor-compress
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: docker-import = /usr/bin/packer-post-processor-docker-import
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: docker-push = /usr/bin/packer-post-processor-docker-push
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: docker-save = /usr/bin/packer-post-processor-docker-save
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: docker-tag = /usr/bin/packer-post-processor-docker-tag
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: vagrant = /usr/bin/packer-post-processor-vagrant
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: vagrant-cloud = /usr/bin/packer-post-processor-vagrant-cloud
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: vsphere = /usr/bin/packer-post-processor-vsphere
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: ansible-local = /usr/bin/packer-provisioner-ansible-local
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: chef-client = /usr/bin/packer-provisioner-chef-client
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: chef-solo = /usr/bin/packer-provisioner-chef-solo
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: file = /usr/bin/packer-provisioner-file
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: puppet-masterless = /usr/bin/packer-provisioner-puppet-masterless
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: puppet-server = /usr/bin/packer-provisioner-puppet-server
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: salt-masterless = /usr/bin/packer-provisioner-salt-masterless
2014/12/16 05:41:53 [DEBUG] Discoverd plugin: shell = /usr/bin/packer-provisioner-shell
2014/12/16 05:41:53 Detected home directory from env var: /home/vagrant
2014/12/16 05:41:53 Detected home directory from env var: /home/vagrant
2014/12/16 05:41:53 Attempting to open config file: /home/vagrant/.packerconfig
2014/12/16 05:41:53 File doesn't exist, but doesn't need to. Ignoring.
2014/12/16 05:41:53 Packer config: &{DisableCheckpoint:false DisableCheckpointSignature:false PluginMinPort:10000 PluginMaxPort:25000 Builders:map[digitalocean:/usr/bin/packer-builder-digitalocean vmware-vmx:/usr/bin/packer-builder-vmware-vmx parallels-pvm:/usr/bin/packer-builder-parallels-pvm qemu:/usr/bin/packer-builder-qemu amazon-chroot:/usr/bin/packer-builder-amazon-chroot amazon-instance:/usr/bin/packer-builder-amazon-instance docker:/usr/bin/packer-builder-docker googlecompute:/usr/bin/packer-builder-googlecompute openstack:/usr/bin/packer-builder-openstack parallels-iso:/usr/bin/packer-builder-parallels-iso virtualbox-iso:/usr/bin/packer-builder-virtualbox-iso virtualbox-ovf:/usr/bin/packer-builder-virtualbox-ovf vmware-iso:/usr/bin/packer-builder-vmware-iso amazon-ebs:/usr/bin/packer-builder-amazon-ebs null:/usr/bin/packer-builder-null] PostProcessors:map[vagrant-cloud:/usr/bin/packer-post-processor-vagrant-cloud vsphere:/usr/bin/packer-post-processor-vsphere atlas:/usr/bin/packer-post-processor-atlas docker-push:/usr/bin/packer-post-processor-docker-push docker-tag:/usr/bin/packer-post-processor-docker-tag vagrant:/usr/bin/packer-post-processor-vagrant compress:/usr/bin/packer-post-processor-compress docker-import:/usr/bin/packer-post-processor-docker-import docker-save:/usr/bin/packer-post-processor-docker-save] Provisioners:map[shell:/usr/bin/packer-provisioner-shell ansible-local:/usr/bin/packer-provisioner-ansible-local chef-client:/usr/bin/packer-provisioner-chef-client chef-solo:/usr/bin/packer-provisioner-chef-solo file:/usr/bin/packer-provisioner-file puppet-masterless:/usr/bin/packer-provisioner-puppet-masterless puppet-server:/usr/bin/packer-provisioner-puppet-server salt-masterless:/usr/bin/packer-provisioner-salt-masterless]}
2014/12/16 05:41:53 Setting cache directory: /vagrant/packer_cache
2014/12/16 05:41:53 Reading template: simple.json
2014/12/16 05:41:53 Detected home directory from env var: /home/vagrant
2014/12/16 05:41:53 Creating build: docker
2014/12/16 05:41:53 Loading builder: docker
2014/12/16 05:41:53 Creating plugin client for path: /usr/bin/packer-builder-docker
2014/12/16 05:41:53 Starting plugin: /usr/bin/packer-builder-docker []string{"/usr/bin/packer-builder-docker"}
2014/12/16 05:41:53 Waiting for RPC address for: /usr/bin/packer-builder-docker
2014/12/16 05:41:53 packer-builder-docker: 2014/12/16 05:41:53 Plugin minimum port: 10000
2014/12/16 05:41:53 packer-builder-docker: 2014/12/16 05:41:53 Plugin maximum port: 25000
2014/12/16 05:41:53 packer-builder-docker: 2014/12/16 05:41:53 Plugin address: unix /tmp/packer-plugin546209489
2014/12/16 05:41:53 packer-builder-docker: 2014/12/16 05:41:53 Waiting for connection...
2014/12/16 05:41:53 packer-builder-docker: 2014/12/16 05:41:53 Serving a plugin connection...
2014/12/16 05:41:53 Loading provisioner: shell
2014/12/16 05:41:53 Creating plugin client for path: /usr/bin/packer-provisioner-shell
2014/12/16 05:41:53 Starting plugin: /usr/bin/packer-provisioner-shell []string{"/usr/bin/packer-provisioner-shell"}
2014/12/16 05:41:53 Waiting for RPC address for: /usr/bin/packer-provisioner-shell
2014/12/16 05:41:53 packer-provisioner-shell: 2014/12/16 05:41:53 Plugin minimum port: 10000
2014/12/16 05:41:53 packer-provisioner-shell: 2014/12/16 05:41:53 Plugin maximum port: 25000
2014/12/16 05:41:53 packer-provisioner-shell: 2014/12/16 05:41:53 Plugin address: unix /tmp/packer-plugin971810863
2014/12/16 05:41:53 packer-provisioner-shell: 2014/12/16 05:41:53 Waiting for connection...
2014/12/16 05:41:53 packer-provisioner-shell: 2014/12/16 05:41:53 Serving a plugin connection...
2014/12/16 05:41:53 Loading provisioner: file
2014/12/16 05:41:53 Creating plugin client for path: /usr/bin/packer-provisioner-file
2014/12/16 05:41:53 Starting plugin: /usr/bin/packer-provisioner-file []string{"/usr/bin/packer-provisioner-file"}
2014/12/16 05:41:53 Waiting for RPC address for: /usr/bin/packer-provisioner-file
2014/12/16 05:41:53 packer-provisioner-file: 2014/12/16 05:41:53 Plugin minimum port: 10000
2014/12/16 05:41:53 packer-provisioner-file: 2014/12/16 05:41:53 Plugin maximum port: 25000
2014/12/16 05:41:53 packer-provisioner-file: 2014/12/16 05:41:53 Plugin address: unix /tmp/packer-plugin052174737
2014/12/16 05:41:53 packer-provisioner-file: 2014/12/16 05:41:53 Waiting for connection...
2014/12/16 05:41:53 packer-provisioner-file: 2014/12/16 05:41:53 Serving a plugin connection...
2014/12/16 05:41:53 Loading provisioner: shell
2014/12/16 05:41:53 Creating plugin client for path: /usr/bin/packer-provisioner-shell
2014/12/16 05:41:53 Starting plugin: /usr/bin/packer-provisioner-shell []string{"/usr/bin/packer-provisioner-shell"}
2014/12/16 05:41:53 Waiting for RPC address for: /usr/bin/packer-provisioner-shell
2014/12/16 05:41:53 packer-provisioner-shell: 2014/12/16 05:41:53 Plugin minimum port: 10000
2014/12/16 05:41:53 packer-provisioner-shell: 2014/12/16 05:41:53 Plugin maximum port: 25000
2014/12/16 05:41:53 packer-provisioner-shell: 2014/12/16 05:41:53 Plugin address: unix /tmp/packer-plugin161242282
2014/12/16 05:41:53 packer-provisioner-shell: 2014/12/16 05:41:53 Waiting for connection...
2014/12/16 05:41:53 packer-provisioner-shell: 2014/12/16 05:41:53 Serving a plugin connection...
2014/12/16 05:41:53 ui: docker output will be in this color.
docker output will be in this color.
2014/12/16 05:41:53 ui: 
2014/12/16 05:41:53 Build debug mode: false
2014/12/16 05:41:53 Force build: false
2014/12/16 05:41:53 Preparing build: docker

2014/12/16 05:41:53 Waiting on builds to complete...
2014/12/16 05:41:53 Starting build run: docker
2014/12/16 05:41:53 Running builder: docker
2014/12/16 05:41:53 ui: ==> docker: Creating a temporary directory for sharing data...
2014/12/16 05:41:53 ui: ==> docker: Pulling Docker image: ubuntu
2014/12/16 05:41:53 packer-builder-docker: 2014/12/16 05:41:53 Executing: /usr/bin/docker [pull ubuntu]
==> docker: Creating a temporary directory for sharing data...
==> docker: Pulling Docker image: ubuntu
2014/12/16 05:41:54 ui:     docker: ubuntu:latest: The image you are pulling has been verified
    docker: ubuntu:latest: The image you are pulling has been verified
2014/12/16 05:41:54 ui:     docker: 511136ea3c5a: Already exists
2014/12/16 05:41:54 ui:     docker: 01bf15a18638: Already exists
    docker: 511136ea3c5a: Already exists
    docker: 01bf15a18638: Already exists
2014/12/16 05:41:54 ui:     docker: 30541f8f3062: Already exists
    docker: 30541f8f3062: Already exists
2014/12/16 05:41:54 ui:     docker: e1cdf371fbde: Already exists
2014/12/16 05:41:54 ui:     docker: 9bd07e480c5b: Already exists
2014/12/16 05:41:54 ui:     docker: Status: Image is up to date for ubuntu:latest
2014/12/16 05:41:54 ui: ==> docker: Starting docker container...
2014/12/16 05:41:54 ui:     docker: Run command: docker run -v /tmp/packer-docker046028796:/packer-files -d -i -t ubuntu /bin/bash
    docker: e1cdf371fbde: Already exists
    docker: 9bd07e480c5b: Already exists
    docker: Status: Image is up to date for ubuntu:latest
==> docker: Starting docker container...
2014/12/16 05:41:54 packer-builder-docker: 2014/12/16 05:41:54 Starting container with args: [run -v /tmp/packer-docker046028796:/packer-files -d -i -t ubuntu /bin/bash]
2014/12/16 05:41:54 packer-builder-docker: 2014/12/16 05:41:54 Waiting for container to finish starting
    docker: Run command: docker run -v /tmp/packer-docker046028796:/packer-files -d -i -t ubuntu /bin/bash
2014/12/16 05:41:54 ui:     docker: Container ID: f436f57f8467caeffdb3e4041bf3e87e489eb1ea64851b12a2815e0f427a4a78
2014/12/16 05:41:54 packer-builder-docker: 2014/12/16 05:41:54 Running the provision hook
2014/12/16 05:41:54 ui: ==> docker: Provisioning with shell script: /tmp/packer-shell756415938
2014/12/16 05:41:54 packer-provisioner-shell: 2014/12/16 05:41:54 Opening /tmp/packer-shell756415938 for reading
2014/12/16 05:41:54 packer-provisioner-shell: 2014/12/16 05:41:54 [INFO] 19 bytes written for 'uploadData'
2014/12/16 05:41:54 [INFO] 19 bytes written for 'uploadData'
2014/12/16 05:41:54 packer-builder-docker: 2014/12/16 05:41:54 Executing in container f436f57f8467caeffdb3e4041bf3e87e489eb1ea64851b12a2815e0f427a4a78: "(cp /packer-files/upload842973995 /tmp/script.sh) >/packer-files/cmd041231502 2>&1; echo $? >/packer-files/cmd041231502-exit"
    docker: Container ID: f436f57f8467caeffdb3e4041bf3e87e489eb1ea64851b12a2815e0f427a4a78
==> docker: Provisioning with shell script: /tmp/packer-shell756415938
2014/12/16 05:41:56 packer-provisioner-shell: 2014/12/16 05:41:56 Retryable error: Error uploading script: Upload failed with non-zero exit status: 1
2014/12/16 05:41:58 packer-provisioner-shell: 2014/12/16 05:41:58 [INFO] 19 bytes written for 'uploadData'
2014/12/16 05:41:58 [INFO] 19 bytes written for 'uploadData'
2014/12/16 05:41:58 packer-builder-docker: 2014/12/16 05:41:58 Executing in container f436f57f8467caeffdb3e4041bf3e87e489eb1ea64851b12a2815e0f427a4a78: "(cp /packer-files/upload805882517 /tmp/script.sh) >/packer-files/cmd043226864 2>&1; echo $? >/packer-files/cmd043226864-exit"
2014/12/16 05:42:00 packer-provisioner-shell: 2014/12/16 05:42:00 Retryable error: Error uploading script: Upload failed with non-zero exit status: 1
2014/12/16 05:42:02 packer-provisioner-shell: 2014/12/16 05:42:02 [INFO] 19 bytes written for 'uploadData'
2014/12/16 05:42:02 [INFO] 19 bytes written for 'uploadData'
2014/12/16 05:42:02 packer-builder-docker: 2014/12/16 05:42:02 Executing in container f436f57f8467caeffdb3e4041bf3e87e489eb1ea64851b12a2815e0f427a4a78: "(cp /packer-files/upload457094543 /tmp/script.sh) >/packer-files/cmd276040098 2>&1; echo $? >/packer-files/cmd276040098-exit"
2014/12/16 05:42:04 packer-provisioner-shell: 2014/12/16 05:42:04 Retryable error: Error uploading script: Upload failed with non-zero exit status: 1
2014/12/16 05:42:06 packer-provisioner-shell: 2014/12/16 05:42:06 [INFO] 19 bytes written for 'uploadData'
2014/12/16 05:42:06 [INFO] 19 bytes written for 'uploadData'
2014/12/16 05:42:06 packer-builder-docker: 2014/12/16 05:42:06 Executing in container f436f57f8467caeffdb3e4041bf3e87e489eb1ea64851b12a2815e0f427a4a78: "(cp /packer-files/upload265236121 /tmp/script.sh) >/packer-files/cmd263955236 2>&1; echo $? >/packer-files/cmd263955236-exit"
2014/12/16 05:42:08 packer-provisioner-shell: 2014/12/16 05:42:08 Retryable error: Error uploading script: Upload failed with non-zero exit status: 1
2014/12/16 05:42:10 [INFO] 19 bytes written for 'uploadData'
2014/12/16 05:42:10 packer-provisioner-shell: 2014/12/16 05:42:10 [INFO] 19 bytes written for 'uploadData'
2014/12/16 05:42:10 packer-builder-docker: 2014/12/16 05:42:10 Executing in container f436f57f8467caeffdb3e4041bf3e87e489eb1ea64851b12a2815e0f427a4a78: "(cp /packer-files/upload876663347 /tmp/script.sh) >/packer-files/cmd882352118 2>&1; echo $? >/packer-files/cmd882352118-exit"
 2014/12/16 05:42:12 packer-provisioner-shell: 2014/12/16 05:42:12 Retryable error: Error uploading script: Upload failed with non-zero exit status: 1
```

I'm stuck here... I can easily reproduce this many times over. Any ideas whats going on?

**Comment by sethvargo** [comment]
I believe this is a duplicate of #1752. Are you able to reproduce this _without_ using Vagrant? If so, that will confirm this is really just #1752. I

**Comment by mitchellh** [comment]
Agreed its a dup, fixed.
